emulator.py

```diff
@@ -8,7 +8,9 @@
 
 import numpy as np
 import os
+from collections import defaultdict
 import sklearn
+import pickle
 from sklearn.neural_network import MLPRegressor
 from sklearn import preprocessing
 from sklearn.preprocessing import StandardScaler
```

Comment:
I don't like the organization of the import functions, so if we have time, I would like to change them. It should not make any difference besides being tidier and following the [PEP 8 convention](https://www.python.org/dev/peps/pep-0008/#imports). Again, this is not to be changed now.